### DISCUSSION TITLE
Tracking down where the vibrations, and video size come from

### DISCUSSION THREAD

**Discussion by kageurufu** [kageurufu/steamdeck_startup_animations#1] [General]
So, I've been back-tracking what makes the vibrations and lays out the video page, and found (all files relative to `~/.steam/steam/steamui`)

## `library.js`

This is pretty basically obfuscated, not positive which obfuscator they used but I'm finding some neat bits and bobs in here. But the important bits are `tVvj` (which I'm calling "BootManager"), which renders out `FMQ2` ("BootAnimation"). `FMQ2` creates a preloader `img` tag, and a `video` tag. 

## `library.css`

In here, I found `.steamdeckbootupthrobber_Container_j78ib video`, which sizes the video element to 300x300px. 
Editing this to 1280x800px fills the screen with the video on reboot! (I removed useless characters from the end of file, in the sourcemap comment to match the original filesize).

## The problem

Every boot now, I get a long boot with the whole "Checking for available updates", "Verifying Installation", "Installing Update". I honestly don't remember if this was happening before I edited library.css, but... I have working full screen boot animations now.

I uploaded the original and edited library.css to https://gist.github.com/kageurufu/10481677718d76daa8135869ac6bb640

I scrapped out this bit of bash scripting to automatically make the edit, but it's not tested. Worst case steam *should* just replace the library.css and you don't have a full screen video, but... who knows

```sh
cp ~/.steam/steam/steamui/css/library.css ~/.steam/steam/steamui/css/library.css.original
sed -i 's/video{flex-grow:0;width:300px;height:300px;z-index:10}/video{flex-grow:0;width:1280px;height:800px;z-index:10}/' ~/.steam/steam/steamui/css/library.css
truncate -r ~/.steam/steam/steamui/css/library.css.original ~/.steam/steam/steamui/css/library.css
```

**Comment by kozak-codes** [comment]
Note that this changes every login and any time steam updates something in the SCSS file for library.css the CSS class names will change (the last 5 characters change randomly in the CSS based on whatever SCSS file they are using and the application because that's the way SCSS works) so you can't rely on `j78ib` to be the same every update. In addition, this will likely get overwritten every time you update. 

#7 sets the width/height to 100% on boot and doesn't rely on library.css. I also added some whitespace so that the replaced characters are the exact same length as the original characters. (we lucked out that 100% is less than 300px lol, otherwise we would have had to find some hacks)

This should permanently fix this so long as steam doesn't change the CSS for this particular selector. It also makes this a bit easier to use.
